specs/DesignDiscussions/InputLabels.md

```diff
@@ -0,0 +1,279 @@
+
+ - Responsiveness: One direction for the future of cards involves re-flowing the layout of a card based on the size of the display. Knowing that a particular label goes with a particular input may help us ensure that semantically related elements stay together on reflow.
+
+ ## Comparisons
+For guidance, consider how labels and inputs are associated in XAML and HTML:
+
+### XAML
+#### Header
+ Xaml TextBox has a “header” property, which is drawn above the text box. Clicking on the header puts focus in the TextBox. Narrator reads the header when the TextBox is selected.
+Here is an example:
+
+ ``` xml
+ <TextBox Header="This is a header"/>
+ ```
+
+![img](assets/InputLabels/XamlHeader.png)
+
+
+Or with a formatted header:
+``` xml
+<TextBox>
+	<TextBox.Header>
+    	<TextBlock  Text="Header" Foreground="red"></TextBlock>
+	</TextBox.Header>
+</TextBox>
+```
+![img](assets/InputLabels/XamlHeaderFormat.png)
+
+Other control types such as Images can also be used as TextBox labels.
+
+#### AutomationProperties
+ Xaml also has an AutomationProperties.LabeledBy attached property for accessibility purposes. This property is used by all elements (not just inputs). If you use it to mark a proceeding text block as the label, it is read the same as a header by narrator, but does not have the click to select behavior of header. 
+
+ ```xml
+<TextBlock x:Name="inputLabel">This is a text block label</TextBlock>
+<TextBox AutomationProperties.LabeledBy="{Binding ElementName=inputLabel}"/> 
+```
+ ![img](assets/InputLabels/XamlLabeledBy.png)
+
+### HTML
+#### Label
+HTML has label tag which can be used to label inputs. The input can either be placed inside the label, or referenced with “for”:
+
+``` HTML
+  <label>
+  Please enter your name:
+  <input id="nameInput">
+  </label>
+```
+
+Or
+``` HTML
+  <label for="nameInput"> Please enter your name:</label>
+  <input id="nameInput">
+```
+ 
+This results in the label being associated with the input both for interactivity and accessibility.
+
+#### Aria
+ HTML also has aria-label and aria-labelledby. These properties are used for accessibility purposes. As in the Xaml AutomationProperties.LabeledBy property, this labels the input for accessibility purposes, but does not change the interactivity model: clicking on the label does not select the input.
+
+## Adaptive Card Schema Options for Semantic Labels
+Below are two possible options for schemas associating labels and inputs. I believe option 1 is the better option, although option 2 allows the additional scenario of having a label separate from the input.
+
+### Option 1 – Label property on inputs
+
+One option would be to make the label text a property of the input itself. This is analogous to the XAML header property discussed above.
+
+``` json
+{
+	"type": "AdaptiveCard",
+	"version": "1.0",
+	"body": [
+		{
+			"type": "Input.Text",
+			"id": "nameInput",
+			"label": "Please enter your name:",
```

Comment:
>label [](start = 4, length = 5)

Comment in the room from Matt: This has a bad back compat story in that the label would be *lost* on earlier versions. That said, the templating library may provide a compatibility layer in the future.

Fallback doesn't work for this scenario, but we could use requires. Even requires doesn't help pre 1.2.

Reply:
A new element type (LabeledInput) would make the fallback story better.

---
In reply to: [347039377](https://github.com/microsoft/AdaptiveCards/pull/3618#discussion_r347039377) [](ancestors = 347039377)

Reply:
@paulcam206 @matthidinger Added back compat sections to each option. I realized that we don't necessarily need a new requires type, the author could just require adaptive cards 1.3. 

Also added a third option to capture @paulcam206's idea of using a new type which includes both the input and the label.

---
In reply to: [347039935](https://github.com/microsoft/AdaptiveCards/pull/3618#discussion_r347039935) [](ancestors = 347039935,347039377)